src/main/groovy/com/codebrig/arthur/observe/structure/filter/conditional/ElseConditionalFilter.groovy

```diff
@@ -17,20 +18,32 @@ import com.google.common.collect.Sets
  */
 class ElseConditionalFilter extends StructureFilter<ElseConditionalFilter, Void> {
 
-    private static final Set<String> conditionalTypes = new HashSet<>()
-    static {
-        conditionalTypes.add("If") //python
-        conditionalTypes.add("IfStmt") //go
-        conditionalTypes.add("IfStatement") //java, javascript
-    }
+    private final MultiFilter elseConditionalFilter
+
     private final Set<Integer> elseNodeIdentities = Sets.newConcurrentHashSet()
 
+    ElseConditionalFilter() {
+        super()
+        this.elseConditionalFilter = createElseConditionalFilter()
+    }
+
+    private static createElseConditionalFilter() {
+        MultiFilter elseStatementFilter = MultiFilter.matchAll(
+                new RoleFilter("IF"), new RoleFilter("STATEMENT")
+        )
+        MultiFilter elseExpressionFilter = MultiFilter.matchAll(
+                new RoleFilter("IF"), new RoleFilter("EXPRESSION")
+        )
+        return MultiFilter.matchAny(elseStatementFilter, elseExpressionFilter)
+    }
+
     @Override
     boolean evaluate(SourceNode node) {
         if (node == null) {
             return false
         }
-        if (node.internalType in conditionalTypes) {
+        boolean result = this.elseConditionalFilter.evaluate(node)
+        if (result) {
             if (node.language == SourceLanguage.Python) {
```

Comment:
There is a language check here. The point of using the semantic roles as the predicate is to get rid of these language checks. Please remove these.